source/features/rgh-improve-new-issue-form.tsx

```diff
@@ -5,6 +5,27 @@ import * as pageDetect from 'github-url-detection';
 import features from '.';
 import clearCacheHandler from '../helpers/clear-cache-handler';
 import {isRefinedGitHubRepo} from '../github-helpers';
+import {expectToken, expectTokenScope} from '../github-helpers/api';
+
+async function validateToken(): Promise<void> {
+	try {
+		await expectToken();
+	} catch {
+		throw new Error('Refined GitHub needs a Personal Access Token to work properly.');
+	}
+
+	try {
+		await expectTokenScope('repo');
+	} catch {
+		throw new Error('Your Personal Access Token is valid but missing the `repo` scope. Many features will not work.');
+	}
+
+	try {
+		await expectTokenScope('delete_repo');
+	} catch {
+		throw new Error('Your Personal Access Token is valid but missing the `delete_repo` scope. The `quick-repo-deletion` feature will not work.');
+	}
```

Comment:
Not sure if it's worth having that precise check

Reply:
I don’t think so, I think the regular `expectTokenScope` error is clear enough. We can just append "[Some features](#the search link we have in the options page) require it"

Reply:
I don't think we need this `quick-repo-deletion` special message either, the feature should already be clear about this when failing.